back/engines/commercial/analytics/spec/services/analytics/query_runner_service_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'query'
+
+describe Analytics::QueryRunnerService do
+  describe 'run' do
+    it 'return the ID field for each post' do
+      ideas = create_list(:idea, 5)
+      create(:dimension_type_idea)
+      query_param = ActionController::Parameters.new(fact: 'post', fields: 'id')
+      query = Analytics::Query.new(query_param)
+
+      runner = described_class.new
+      results = runner.run(query)
+      expect(results).to match_array(ideas.map { |idea| { 'id' => idea.id } })
+    end
+
+    it 'return groups with aggregations' do
+      create_list(:idea, 5)
+      create_list(:initiative, 5)
+      create(:dimension_type_idea)
+      create(:dimension_type_initiative)
+      create_list(:vote, 10)
```

Comment:
This line is creating 10 ideas. Maybe we could decrease the number and remove `create_list(:idea, 5)` which does not change anything here.

Reply:
It would be nice to test a mix of votes on both initiatives and ideas.